src/test/java/org/prebid/server/bidder/medianet/MedianetBidderTest.java

```diff
@@ -38,8 +46,121 @@ public void setup() {
 
     @Test
     public void creationShouldFailOnInvalidEndpointUrl() {
-        assertThatIllegalArgumentException()
-                .isThrownBy(() -> new MedianetBidder("invalid_url", jacksonMapper));
+        assertThatIllegalArgumentException().isThrownBy(() -> new MedianetBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestShouldReturnErrorIfImpressionListSizeIsZero() {
+        //given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(emptyList())
+                .build();
+
+        //when
+        final Result<List<HttpRequest<BidRequest>>> result = medianetBidder.makeHttpRequests(bidRequest);
+
+        //then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors()).hasSize(1);
+        assertThat(result.getErrors())
+                .allMatch(error -> error.getType() == bad_input
+                        && error.getMessage().equals("No valid impressions in the bid request"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnBannerBidIfBannerIsPresentInRequestImp() throws JsonProcessingException {
+        //given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(BidRequest.builder()
+                        .imp(singletonList(Imp.builder().id("123").banner(Banner.builder().build()).build()))
+                        .build(),
+                mapper.writeValueAsString(
+                        givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        //when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        //then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1);
+        assertThat(result.getValue())
```

Comment:
Can be simplified with `containsExactly`, also please check for similar occurences.

Reply:
So, I think it is not better way because need write full error message.
In my case : _"Failed to decode: Unrecognized token 'invalid': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')\n at [Source: (String)\"invalid response\"; line: 1, column: 8]"_

Reply:
You can get around this by using `.allSatisfy()` + `startsWith()` in combination, to check only for "Failed to decode".